sql/parser/overload.go

```diff
@@ -182,6 +182,10 @@ func typeCheckOverloadedExprs(
 	// Hold the resolved type expressions of the provided exprs, in order.
 	typedExprs := make([]TypedExpr, len(exprs))
 
+	// Split provided expressions into three groups:
+	// - Arguments
```

Comment:
We might as well keep it consistent until I clean up the rest of the vocabulary used here. I'll make the change after https://github.com/cockroachdb/cockroach/pull/6983 goes through.